intelmq/bots/parsers/blueliv/parser_crimeserver.py

```diff
@@ -25,14 +27,25 @@ def process(self):
 
         for item in json.loads(raw_report):
             event = self.new_event(report)
+            tor_node = False
+            if 'type' in item:
+                event.add('classification.type', TYPES[item['type']])
+                if item['type'] == 'TOR_IP':
+                    event.add('source.tor_node', True)
+                    tor_node = True
             if 'url' in item:
-                event.add('source.url', item['url'])
+                # crimeserver reports tor ips in url as well, skip those
+
+                if not tor_node:
+                    event.add('source.url', item['url'])
+                else:
+                    valid_ip = IPAddress.is_valid(item['url'])
```

Comment:
changed the logic by using `raise_failure`, the flg is set for tor_node and `raise_failure` set t False for tor_nodes. Much simpler code.